### PR TITLE
GG-35225 Enable keepBinary by default for sqlline

### PR DESCRIPTION
**Pull request by tledkov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java [hunk 1]
```diff
@@ -1027,6 +1028,12 @@ else if (res.status() != ClientListenerResponse.STATUS_SUCCESS)
 
                     throw e;
                 }
+                catch (BinaryObjectException e) {
+                    if (LOG.isLoggable(Level.FINE))
+                        LOG.log(Level.FINE, "Exception during sending an sql request.", e);
```
Comment (ygerzhedovich) [inline modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:1033]: Maybe log possible solution here?

Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:1033]: Not catch an idea. Add to message info about binary mode?

Comment (ygerzhedovich) [reply modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:1033]: Hmmm, may be you are right and add info to exception will be more good solution. Seems will be good to add solutions like a ''Probably you use specific non JDBC types, in this case try to set keepBinary=true for connection"

Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java]: Fixed.

**Pull request merged**